lucille-plugins/lucille-api/src/main/java/com/kmwllc/lucille/endpoints/LucilleResource.java

```diff
@@ -1,61 +1,211 @@
 package com.kmwllc.lucille.endpoints;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.kmwllc.lucille.AuthHandler;
+import com.kmwllc.lucille.core.RunDetails;
 import com.kmwllc.lucille.core.RunnerManager;
-import com.kmwllc.lucille.objects.RunStatus;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
 import io.dropwizard.auth.Auth;
 import io.dropwizard.auth.PrincipalImpl;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.ws.rs.Consumes;
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.POST;
-
 import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
 import jakarta.ws.rs.Produces;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.Response;
-import java.util.Optional;
 
 /**
  * Lucille Admin API Endpoint:
  *
- * - endpoint: '/lucille'
- *   - GET: obtain lucille run status
- *   - POST: Kick off new lucille run, if one is not already running
  */
-@Path("/lucille")
+@Path("/v1")
 @Produces(MediaType.APPLICATION_JSON)
 public class LucilleResource {
 
+  private static final Logger log = LoggerFactory.getLogger(RunnerManager.class);
+
   private final RunnerManager runnerManager;
 
-  public LucilleResource(RunnerManager runnerManager) {
+  private final AuthHandler authHandler;
+
+  public LucilleResource(RunnerManager runnerManager, AuthHandler authHandler) {
     this.runnerManager = runnerManager;
+    this.authHandler = authHandler;
+  }
+
+  @POST
+  @Tag(name = "Config")
+  @Path("/config")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Create a new Lucille config to be run later",
+      description = "Creates a new Lucille run with the specified configuration")
+  public Response createRun(@Parameter(hidden = true) @Auth Optional<PrincipalImpl> user,
```

Comment:
rename method to createConfig

Reply:
This comment is about the java method `public Response createRun(` not the `POST config` endpoint name. Yes, we agreed to POST config in the meeting and that's all good, but it looks like the java method name here is left over from before. Also, the Operation summary "Create a new Lucille config to be run later" is accurate. But the Operation description "Creates a new Lucille run with the specified configuration" looks like it should be updated.

Reply:
:facepalm: Thanks, of course, that makes sense.
fixed